VAMobile/android/app/src/main/java/gov/va/mobileapp/native_modules/CustomTabsIntentModule.kt

```diff
@@ -84,13 +86,27 @@ class CustomTabsIntentModule(private val context: ReactApplicationContext) :
                             }
                             .build()
 
-            // Check default browser to prevent Firefox login issue (Android only)
-            val browserIntent = Intent("android.intent.action.VIEW", Uri.parse("https://"));
-            val resolveInfo = context.packageManager.resolveActivity(browserIntent, PackageManager.MATCH_DEFAULT_ONLY);
-            val packageName = resolveInfo?.activityInfo?.packageName;
-            if (packageName != null && packageName.contains("firefox")) {
-                // Default browser is Firefox. Need flag for login to succeed
-                customTabsIntent.intent.addFlags(FLAG_ACTIVITY_NEW_TASK)
+            // Get full list of installed packages that handle https URLs
+            // https://developer.chrome.com/docs/android/custom-tabs/howto-custom-tab-check/
+            val intent = Intent(ACTION_VIEW, Uri.parse("https://www.example.com"))
+            val intentHandlers = context.packageManager.queryIntentActivities(intent, PackageManager.MATCH_ALL)
+            val intentHandlerPackages = intentHandlers.map {
+                it.activityInfo.packageName
+            }.toTypedArray().asList()
+
+            // From the above list, find a package that supports Custom Tabs. If no installed
+            // packages support Custom Tabs, packageName will be set to null here
+            val packageName = CustomTabsClient.getPackageName(context, intentHandlerPackages)
+
+            // If packageName is present, that package supports Custom Tabs, so use it
+            // TODO: Inform user when no installed packages support Custom Tabs
```

Comment:
Did this already have a ticket number to point to? If so, may be good to list.

Reply:
There's no ticket on this one yet. I think there are some open questions on exactly how and when to notify users, in addition to the technical considerations of whether this happens in native code or not. Probably we'd need some input beyond engineers. There's also the wider consideration of whether we continue to refine the custom tabs approach or whether we move towards in-app login